### ISSUE TITLE
Registry authentication fails when secret contains only .dockercfg

### ISSUE THREAD

**Issue by derkoe** [ckotzbauer/sbom-operator#26]
Secrets with `type = "kubernetes.io/dockercfg"` containing the pull secret in the field `.dockercfg` do not work. 

OpenShift uses this type of secret for the internal registry - so, you cannot provide the `type = "kubernetes.io/dockerconfigjson"`.

**Comment by derkoe** [comment]
This is not fixed with that commit since `LoadFromReader` does not use the old format. Only `Load` handles this case (distinguishing by file name).

**Comment by ckotzbauer** [comment]
I did a small research on this: I think the legacy `.dockercfg` is a subset of the newer `.docker/config.json` file which only supports the `auths` section (which is identically to the `auths` section in newer config-files). The code in `LoadFromReader` and `LegacyLoadFromReader` (which is called from `Load` when the old-file is detected) is almost the same regarding the `auths` section. So I think old files should be handled correctly too from `LoadFromReader`.

Can you give me an sanitized example of your `.dockercfg` file?

**Comment by derkoe** [comment]
This is the `.dockercfg`:
```json
{
    "docker.example.com": { "username": "user", "password": "password" }
}
```
This is the `.dockerconfigjson` (where the aut is `user:password` base64 encoded):
```
{
    "auths": { "docker.example.com": { "auth": "dXNlcjpwYXNzd29yZA==" } }
}
```

**Comment by ckotzbauer** [comment]
This should be finally fixed now in 0.4.1 @derkoe. There's a unit-test now to keep this working. However, I don't know if the fact that there's no `auth` property in the `.dockercfg` is really official, as the official docker-code cannot handle an empty `auth` property. I ported the specific loading-logic to the application and modified it to make it work.
